### PR TITLE
enhancement(internal_logs, internal_metrics source): Add test helpers to assert source compliance

### PR DESCRIPTION
**Pull request by neuronull**
Closes: #14129.
Closes: #14130.

Part of  #13995.

- Added test helper assertions
- Added `BytesReceived` event emission to `internal_metrics` in order to comply

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
Sweet!

File: src/internal_events/internal_metrics.rs [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+pub struct InternalMetricsBytesReceived {
+    pub byte_size: usize,
+}
+
+impl InternalEvent for InternalMetricsBytesReceived {
+    fn emit(self) {
+        trace!(
+            message = "Bytes received.",
+            byte_size = %self.byte_size,
+            protocol = "internal",
```
Comment (spencergilbert) [inline src/internal_events/internal_metrics.rs:14]: I'm not 100% on the `protocol` here, perhaps it shouldn't be required in the spec?

**Review by spencergilbert — APPROVED** [review]

File: src/internal_events/internal_metrics.rs (hunk 1 above)
Comment (neuronull) [reply src/internal_events/internal_metrics.rs:14]: Do you mean that you aren't 100% convinced that `protocol` needs to be a parameter here?
I think I was mimicking `InternalLogsBytesReceived`.

Comment (spencergilbert) [reply src/internal_events/internal_metrics.rs:14]: Yeah, I'm not sure `internal` makes sense as a "protocol" but I think it's in the spec for these. May be a place to re-evaluate if it's just noise later on.

**Pull request merged**